test/jdk/java/math/BigDecimal/Constructor.java

```diff
@@ -26,45 +26,57 @@
  * @bug 4259453 8200698
  * @summary Test constructors of BigDecimal
  * @library ..
- * @run testng Constructor
+ * @run junit Constructor
  */
 
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
 import java.math.BigDecimal;
-import org.testng.annotations.Test;
 
 public class Constructor {
-    @Test(expectedExceptions=NumberFormatException.class)
+    @Test
     public void stringConstructor() {
-        BigDecimal bd = new BigDecimal("1.2e");
+        Assertions.assertThrows(NumberFormatException.class, () -> {
+            BigDecimal bd = new BigDecimal("1.2e");
+        });
     }
 
-    @Test(expectedExceptions=NumberFormatException.class)
+    @Test
     public void charArrayConstructorNegativeOffset() {
-        BigDecimal bd = new BigDecimal(new char[5], -1, 4, null);
+        Assertions.assertThrows(NumberFormatException.class, () -> {
+            BigDecimal bd = new BigDecimal(new char[5], -1, 4, null);
+        });
     }
 
-    @Test(expectedExceptions=NumberFormatException.class)
+    @Test
     public void charArrayConstructorNegativeLength() {
-        BigDecimal bd = new BigDecimal(new char[5], 0, -1, null);
+        Assertions.assertThrows(NumberFormatException.class, () -> {
+            BigDecimal bd = new BigDecimal(new char[5], 0, -1, null);
+        });
     }
 
-    @Test(expectedExceptions=NumberFormatException.class)
+    @Test
     public void charArrayConstructorIntegerOverflow() {
-        try {
-            BigDecimal bd = new BigDecimal(new char[5], Integer.MAX_VALUE - 5,
-                6, null);
-        } catch (NumberFormatException nfe) {
-            if (nfe.getCause() instanceof IndexOutOfBoundsException) {
-                throw new RuntimeException
-                    ("NumberFormatException should not have a cause");
-            } else {
-                throw nfe;
+        Assertions.assertThrows(NumberFormatException.class, () -> {
+            try {
+                BigDecimal bd = new BigDecimal(new char[5], Integer.MAX_VALUE - 5,
+                        6, null);
+            } catch (NumberFormatException nfe) {
+                if (nfe.getCause() instanceof IndexOutOfBoundsException) {
+                    throw new RuntimeException
+                            ("NumberFormatException should not have a cause");
+                } else {
+                    throw nfe;
+                }
```

Comment:
If you want, you could use `NumberFormatException nfe = assertThrows(..)` here, then assert the cause. That would avoid mixing JUnit asserts with explicit throw of RuntimeException.

Reply:
It is my plan to review all numerics related tests, rewriting some of those that do not use any framework to use junit in the end.
For now, the priority is to remove the dependencies on testng.